.castor.php

```diff
@@ -0,0 +1,443 @@
+        $io->note('Run "castor infrastructure:generate-certificates --force" to generate new certificates.');
+
+        return;
+    }
+
+    if ($force) {
+        if (file_exists($f = $c['root_dir'] . '/infrastructure/docker/services/router/etc/ssl/certs/cert.pem')) {
+            $io->comment('Removing existing certificates in infrastructure/docker/services/router/etc/ssl/certs/*.pem.');
+            unlink($f);
+        }
+
+        if (file_exists($f = $c['root_dir'] . '/infrastructure/docker/services/router/etc/ssl/certs/key.pem')) {
+            unlink($f);
+        }
+    }
+
+    $finder = new ExecutableFinder();
+    $mkcert = $finder->find('mkcert');
+
+    if ($mkcert) {
+        $pathCaRoot = trim(exec(['mkcert', '-CAROOT'], quiet: true)->getOutput());
+
+        if (!is_dir($pathCaRoot)) {
+            $io->warning('You must have mkcert CA Root installed on your host with "mkcert -install" command.');
+
+            return;
+        }
+
+        exec([
+            'mkcert',
+            '-cert-file', 'infrastructure/docker/services/router/etc/ssl/certs/cert.pem',
+            '-key-file', 'infrastructure/docker/services/router/etc/ssl/certs/key.pem',
+            $c['root_domain'],
+            "*.{$c['root_domain']}",
+            ...$c['extra_domains'],
+        ]);
+
+        $io->success('Successfully generated SSL certificates with mkcert.');
+
+        if ($force) {
+            $io->note('Please restart the infrastructure to use the new certificates with "castor up" or "castor start".');
+        }
+
+        return;
+    }
+
+    exec(['infrastructure/docker/services/router/generate-ssl.sh']);
+
+    $io->success('Successfully generated self-signed SSL certificates in infrastructure/docker/services/router/etc/ssl/certs/*.pem.');
+    $io->comment('Consider installing mkcert to generate locally trusted SSL certificates and run "castor infra:generate-certificates --force".');
+
+    if ($force) {
+        $io->note('Please restart the infrastructure to use the new certificates with "castor up" or "castor start".');
+    }
+}
+
+#[AsTask(description: 'Starts the workers', namespace: 'infra:worker', name: 'start')]
+function workers_start(Context $c)
+{
+    $workers = get_workers($c);
+
+    if (!$workers) {
+        return;
+    }
+
+    exec([
+        'docker',
+        'update',
+        '--restart=unless-stopped',
+        ...$workers,
+    ], quiet: true);
+
+    exec([
+        'docker',
+        'start',
+        ...$workers,
+    ], quiet: true);
+}
+
+#[AsTask(description: 'Stops the workers', namespace: 'infra:worker', name: 'stop')]
+function workers_stop(Context $c)
+{
+    $workers = get_workers($c);
+
+    if (!$workers) {
+        return;
+    }
+
+    exec([
+        'docker',
+        'update',
+        '--restart=no',
+        ...$workers,
+    ]);
+
+    exec([
+        'docker',
+        'stop',
+        ...$workers,
+    ]);
+}
+
+#[AsContext(default: true)]
+function create_default_context(): Context
```

Comment:
I suggest to move this function at the very top as it the one each new project should edit

Reply:
What about having a simple function like `create_default_parameters()`, with the 4/5 first lines of this files, at the top of this files. Then, we use it in this function?